go/vt/vtgate/planbuilder/operators/projection.go

```diff
@@ -201,16 +201,14 @@ func createSimpleProjection(ctx *plancontext.PlanningContext, selExprs []sqlpars
 // been settled. Once they have settled, we know where to push the projection, but if we push too early
 // the projection can end up in the wrong branch of joins
 func (p *Projection) canPush(ctx *plancontext.PlanningContext) bool {
-	if reachedPhase(ctx, subquerySettling) {
-		return true
-	}
+	subQSettled := reachedPhase(ctx, subquerySettling)
 	ap, ok := p.Columns.(AliasedProjections)
 	if !ok {
 		// we can't mix subqueries and unexpanded stars, so we know this does not contain any subqueries
 		return true
 	}
 	for _, projection := range ap {
-		if _, ok := projection.Info.(SubQueryExpression); ok {
+		if _, ok := projection.Info.(SubQueryExpression); ok && !subQSettled {
 			return false
 		}
 	}
```

Comment:
I guess we can pull up the `!subQSettled` check here and skip the whole loop unless `!subQSettled` is true?